modules/core/src/main/java/org/apache/synapse/message/processors/forward/ForwardingJob.java

```diff
@@ -80,20 +85,31 @@ private void configureForwardingJob(JobDataMap jdm) {
                 ScheduledMessageForwardingProcessor.BLOCKING_SENDER);
         processor = (ScheduledMessageForwardingProcessor) jdm.get(
                 ScheduledMessageForwardingProcessor.PROCESSOR_INSTANCE);
+        retryInterval = 1000;
+
+        setParameters(jdm);
+    }
+
+    private void setParameters(JobDataMap jdm) {
         Map<String, Object> parameters = (Map<String, Object>) jdm.get(MessageProcessorConstants.PARAMETERS);
-        maxDeliverAttempts = extractMaxDeliveryAttempts(parameters, processor);
-        isMaxDeliverAttemptDropEnabled = isMaxDeliverAttemptDropEnabled(parameters);
-        retryHttpStatusCodes(parameters);
-        setSequences(parameters);
+        if (parameters != null) {
+            maxDeliverAttempts = extractMaxDeliveryAttempts(parameters, processor);
+            isMaxDeliverAttemptDropEnabled = isMaxDeliverAttemptDropEnabled(parameters);
+            if (parameters.get(ForwardingProcessorConstants.RETRY_INTERVAL) != null) {
+                retryInterval = Integer.parseInt((String) parameters.get(ForwardingProcessorConstants.RETRY_INTERVAL));
```

Comment:
Shouldn't we handle NumberformatException here?

Reply:
Thanks for pointing out. I missed it. I have handle the exception by falling back to default value and also logged the error message.

`2020-04-20 21:35:48,458 [-] [DefaultQuartzScheduler_Worker-1] ERROR ForwardingJob Invalid value for retry.interval switching back to default value
`